var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -261,7 +282,7 @@ class Qt(Package):
     patch("qt51514-oneapi.patch", when="@5.15.14: %oneapi")
 
     # Non-macOS dependencies and special macOS constraints
-    if MACOS_VERSION is None:
+    if MACOS_VERSION is None and not IS_WINDOWS:
```

Comment:
I think it would be clearer to make this a loop over `[linux, freebsd]`, or at least, this is the same concept as the "everywhere-but-windows" loop above, so it's mildly confusing to handle it differently in each case.